sample/Broadcast/README.md

```diff
@@ -0,0 +1,180 @@
+
+```ruby
+ get '/host' do
+    api_key = settings.api_key
+    session_id = settings.session.session_id
+    token = settings.opentok.generate_token(session_id, role: :publisher, initialLayoutClassList: ['focus'])
+
+    erb :host, locals: {
+        apiKey: api_key,
+        sessionId: session_id,
+        token: token,
+        initialBroadcastId: settings.broadcastId,
+        focusStreamId: settings.focusStreamId,
+        initialLayout: settings.broadcastLayout
+    }
+  end
+
+```
+
+This handler simply
+generates the three strings that the client (JavaScript) needs to connect to the session: `api_key`,
+`session_id` and `token`. The `initialBroadcastId` is the broadcast id , `focusStreamId` is the stream id which has the current
+focus and `initialLayout` is the initial layout for the current broadcast which is being started. After the user 
+has connected to the session, they press the
+'Start Broadcast' button, which sends an XHR (or Ajax) request to the <http://localhost:4567/start>
+URL. The route handler for this URL is shown below:
+
+```ruby
+  post '/start' do
+    opts = {
+        :maxDuration => params.key?("maxDuration") ? params[:maxDuration] : 7200,
+        :resolution =>  params[:resolution],
+        :layout => params[:layout],
+        :outputs => {
+            :hls => {}
+        }
+    }
+    broadcast = settings.opentok.broadcasts.create(settings.session.session_id, opts)
+    settings.broadcastId = broadcast.id
+    body broadcast.to_json
+  end
+```
+
+In this handler, `opentok.broadcasts.create` is called with the `session_id` for the broadcasting session
+The optional second argument is a hash which defines optional properties
+for the broadcast. It consists of `maxDuration` of the broadcast, `resolution` and broadcast `layout`.
+In this sample app we only care for `hls` broadcasting hence we only specify that. 
+You can add RTMP servers , if you desire. Please refer Ruby [SDK documentation](https://github.com/opentok/OpenTok-Ruby-SDK) for that. 
+In this case, as in the
+HelloWorld sample app, there is only one session created and it is used here and for the participant
+view. This will trigger the broadcasting to begin. The response sent back to the client's XHR request
+will be the JSON representation of the archive, which is returned from the `to_json()` method. 
+
+You can view the HLS broadcast by going to your Home page in a different tab and clicking on `Broadcast URL` button.
+The code for handling this is as follows:
+```ruby
+  get '/broadcast' do
+    broadcast = settings.opentok.broadcasts.find settings.broadcastId
+    redirect broadcast.broadcastUrls['hls'] if broadcast.status == 'started'
+  end
+```
+The Stop Broadcast code looks like:
+
+```ruby
+  get '/stop/:broadcastId' do
+    broadcast = settings.opentok.broadcasts.stop settings.broadcastId
+    settings.broadcast = nil
+    settings.focusStreamId = ''
+    settings.broadcastLayout = 'bestFit'
```

Comment:
The layout of the screens in the client (web pages) should match the layout of the broadcast. So, the app should only use the `verticalPresentation` and `horizontalPresentation` layouts (not `bestFit`).